pyrdp/core/twisted.py

```diff
@@ -29,4 +34,39 @@ def buildProtocol(self, addr):
         if callable(self.protocol):
             return self.protocol()
         else:
-            return self.protocol
\ No newline at end of file
+            return self.protocol
+
+
+class TransparentClient(tcp.Client):
+    """A TCP client that supports transparent proxying."""
+
+    def createInternetSocket(self):
+        """Overridden"""
+
+        err = None
+        s = socket.socket(self.addressFamily, self.socketType)
+        s.setblocking(0)
+        fdesc._setCloseOnExec(s.fileno())
+
+        try:
+            if not s.getsockopt(socket.SOL_IP, socket.IP_TRANSPARENT):
+                s.setsockopt(socket.SOL_IP, socket.IP_TRANSPARENT, 1)
+        except Exception as e:
+            LOG.error('Failed to establish transparent proxy: %s', e)
+
+        return s  # Maintain non-transparent behavior.
+
+
+class TransparentConnector(tcp.Connector):
+    """A TCP connector which creates TransparentClients."""
+
+    def _makeTransport(self):
+        return TransparentClient(self.host, self.port, self.bindAddress, self, self.reactor)
+
+
+def connectTransparent(host, port, factory, timeout=30, bindAddress=None):
+    from twisted.internet import reactor
```

Comment:
why is the import statement there?

Reply:
I wanted to make sure that importing this file first does not initialize a different reactor than the one installed in `pyrdp-mitm.py`. Since twisted seems to lazily create the reactor. I might be wrong though, but I remember that moving some imports related to twisted earlier caused some exceptions. So I'd rather lazily import reactor the first time this function is called.